samples/igr/igr.c

```diff
@@ -0,0 +1,84 @@
+#include <assert.h>
+#include <xboxkrnl/xboxkrnl.h>
+
+#include "igr.h"
+
+// `connected` parameter was added manually
+static uint32_t fake_XInputGetState(void* device, IGRPadState* state, uint32_t return_value) {
+
+  // Confirm that our code below is exactly what nkpatcher expects
+  extern uint8_t pattern_values[];
+  uint8_t pattern_mask[]      = { 0xFF,0x00,0xFF,0xFF,0x00,0x00,0xFF,0xFF,0xFF,0xFF };
+  uint8_t pattern_reference[] = { 0x8B,   0,0x24,0x0C,   0,   0,0xA3,0x00,0x00,0x00 };
+  for(int i = 0; i < sizeof(pattern_mask); i++) {
+    assert((pattern_values[i] & pattern_mask[i]) == pattern_reference[i]);
+  }
+
+
+  // It hooks KeRaiseIrqlToDpcLevel and the caller must have a specific pattern of bytes.
+  // This assembly block matches that.
+  uint8_t irql;
+  asm volatile("movl $0x55500555, %%edx;\n" // Marker to make it easier to find
```

Comment:
I suppose that marker is for finding that code-part when looking at the compiled result via an hex editor? 
Additional, it bothers me a bit that the comment is behind it, instead of on top of it - as it with all other asm-comments. But thats just my taste 😛

Reply:
Yes; and it's a single line to make it more compact when I want to delete it.